File: appejv-api/internal/models/product.go
package models

import "time"

type Product struct {
	ID             int        `json:"id"`
	Code           string     `json:"code"`
	Name           string     `json:"name"`
	Slug           *string    `json:"slug,omitempty"`
	Unit           *string    `json:"unit,omitempty"`
	Stock          int        `json:"stock"`
	ImageURL       *string    `json:"image_url,omitempty"`
	Price          float64    `json:"price"`
	Category       *string    `json:"category,omitempty"`
	CategoryID     *int       `json:"category_id,omitempty"`
	Description    *string    `json:"description,omitempty"`
	Specifications *string    `json:"specifications,omitempty"`
	DeletedAt      *time.Time `json:"deleted_at,omitempty"`
	CreatedAt      time.Time  `json:"created_at"`
}

type CreateProductRequest struct {
	Code           string  `json:"code" binding:"required"`
	Name           string  `json:"name" binding:"required"`
	Unit           *string `json:"unit"`
	Stock          int     `json:"stock"`
	Price          float64 `json:"price"`
	Category       *string `json:"category"`
	CategoryID     *int    `json:"category_id"`
	Description    *string `json:"description"`
	ImageURL       *string `json:"image_url"`
	Specifications *string `json:"specifications"`
}

type UpdateProductRequest struct {
	Name           *string  `json:"name"`
	Unit           *string  `json:"unit"`
	Stock          *int     `json:"stock"`
	Price          *float64 `json:"price"`
	Category       *string  `json:"category"`
	CategoryID     *int     `json:"category_id"`
	Description    *string  `json:"description"`
	ImageURL       *string  `json:"image_url"`
	Specifications *string  `json:"specifications"`
}

File: appejv-android/app/src/main/java/com/appejv/android/core/data/model/Product.kt
package com.appejv.android.core.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Product(
    @SerialName("id")
    val id: Int,
    
    @SerialName("code")
    val code: String,
    
    @SerialName("name")
    val name: String,
    
    @SerialName("slug")
    val slug: String? = null,
    
    @SerialName("unit")
    val unit: String? = null,
    
    @SerialName("stock")
    val stock: Int,
    
    @SerialName("image_url")
    val imageUrl: String? = null,
    
    @SerialName("price")
    val price: Double,
    
    @SerialName("category")
    val category: String? = null,
    
    @SerialName("category_id")
    val categoryId: Int? = null,
    
    @SerialName("description")
    val description: String? = null,
    
    @SerialName("specifications")
    val specifications: String? = null,
    
    @SerialName("created_at")
    val createdAt: String? = null
)

@Serializable
data class ProductsResponse(
    @SerialName("data")
    val data: List<Product>,
    
    @SerialName("total")
    val total: Int? = null
)
